lib/Models/Workbench.ts

```diff
@@ -34,7 +34,12 @@ export default class Workbench {
     return this._items.map(dereferenceModel);
   }
   set items(items: readonly BaseModel[]) {
-    this._items.spliceWithArray(0, this._items.length, items.slice());
+    const setItems = new Set(items);
+    this._items.spliceWithArray(
```

Comment:
Probably add a comment here to say that we want the workbench items to be unique.

Reply:
good idea, commented